cmd/config.go

```diff
@@ -229,39 +217,45 @@ func migrateConfigToMinioSys() error {
 }
 
 // Initialize and load config from remote etcd or local config directory
-func initConfig() {
+func initConfig(objAPI ObjectLayer) error {
+	if objAPI == nil {
+		return errServerNotInitialized
+	}
+
 	if globalEtcdClient != nil {
 		ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
 		resp, err := globalEtcdClient.Get(ctx, getConfigFile())
 		cancel()
 		if err == nil && resp.Count > 0 {
-			logger.FatalIf(migrateConfig(), "Config migration failed")
+			return migrateConfig()
```

Comment:
with this we'll be losing granular information on what failed during `initConfig`, why not keep `logger.FatalIf()`

Reply:
Structurally its not needed, the change is in line with all other functions for ConfigSys.